### PR TITLE
Make null-terminated string reading more robust

### PR DESCRIPTION
**Pull request by roji**
Also add proper testing infrastructure for the read buffer

/cc @Brar

### CODE REVIEW

File: src/Npgsql/NpgsqlReadBuffer.cs [hunk 1]
```diff
@@ -549,67 +549,60 @@ public Stream GetStream(int len, bool canSeek)
...
-        public string ReadNullTerminatedStringRelaxed() => ReadNullTerminatedString(RelaxedTextEncoding);
+        public string ReadNullTerminatedStringRelaxed()
+            => ReadNullTerminatedString(RelaxedTextEncoding, async: false).GetAwaiter().GetResult();
+
+        public ValueTask<string> ReadNullTerminatedString(bool async, CancellationToken cancellationToken = default)
+            => ReadNullTerminatedString(TextEncoding, async, cancellationToken);
 
         /// <summary>
         /// Seeks the first null terminator (\0) and returns the string up to it. The buffer must already
         /// contain the entire string and its terminator.
```
Comment (Brar) [inline src/Npgsql/NpgsqlReadBuffer.cs:568]: > The buffer must already contain the entire string and its terminator.

I think this isn't true anymore - no?

**Review by Brar — APPROVED** [review]
That really improves the state of arts.
Simpler, more robust - good.

See my minor comment about the comment

File: src/Npgsql/NpgsqlReadBuffer.cs (hunk 1 above)
Comment (roji) [reply src/Npgsql/NpgsqlReadBuffer.cs:568]: Thanks, fixed.

**Comment by roji** [comment]
Merged via ebed71457b1f0312224942a54740bc87ac6eeb44

**Pull request closed without merging**